python/ray/tune/trainable/function_trainable.py

```diff
@@ -519,6 +528,9 @@ def restore_from_object(self, obj):
         checkpoint.to_directory(self.temp_checkpoint_dir)
 
         self.restore(self.temp_checkpoint_dir)
+        # Set tmp restore dir - this directory will be deleted once a new checkpoint
+        # is written or set.
+        self._status_reporter._tmp_restore_dir = self.temp_checkpoint_dir
```

Comment:
Instead of keeping the tmp restore dir around, is it possible to just check if the previous checkpoint was a tmp checkpoint during `set_checkpoint` (using `FuncCheckpointUtil.is_temp_checkpoint_dir`), then deleting if so?

Like:

```python
def set_checkpoint(checkpoint):
    if last checkpoint was a tmp ckpt, delete
    last_checkpoint = checkpoint
```

This can keep the logic contained in status reporter. Lmk what you think

Reply:
Great point, updated